lib/post_destroyer.rb

```diff
@@ -196,6 +198,17 @@ def trash_public_post_actions
     Post.with_deleted.where(id: @post.id).update_all(Hash[*f.flatten])
   end
 
+  def recover_public_post_actions(post_deleted_at)
+    PostAction.publics
```

Comment:
This is going to generate N+1 queries. 

Why not do `.update_all(deleted_at: nil, deleted_by_id: nil)` instead of the `.find_each` loop?